Modules/Filtering/ImageFeature/include/itkLaplacianSharpeningImageFilter.hxx

```diff
@@ -35,136 +33,53 @@ LaplacianSharpeningImageFilter<TInputImage, TOutputImage>::PrintSelf(std::ostrea
   os << indent << "UseImageSpacing: " << (m_UseImageSpacing ? "On" : "Off") << std::endl;
 }
 
-template <typename TInputImage, typename TOutputImage>
-void
-LaplacianSharpeningImageFilter<TInputImage, TOutputImage>::GenerateInputRequestedRegion()
-{
-  // call the superclass' implementation of this method. This should
-  // copy the output requested region to the input requested region
-  Superclass::GenerateInputRequestedRegion();
-
-  // get pointers to the input and output
-  InputImagePointer inputPtr = const_cast<TInputImage *>(this->GetInput());
-
-  if (!inputPtr)
-  {
-    return;
-  }
-
-  // Build an operator so that we can determine the kernel size
-  LaplacianOperator<RealType, ImageDimension> oper;
-  oper.CreateOperator();
-
-  // get a copy of the input requested region (should equal the output
-  // requested region)
-  typename TInputImage::RegionType inputRequestedRegion;
-  inputRequestedRegion = inputPtr->GetRequestedRegion();
-
-  // pad the input requested region by the operator radius
-  inputRequestedRegion.PadByRadius(oper.GetRadius());
-
-  // crop the input requested region at the input's largest possible region
-  if (inputRequestedRegion.Crop(inputPtr->GetLargestPossibleRegion()))
-  {
-    inputPtr->SetRequestedRegion(inputRequestedRegion);
-    return;
-  }
-  else
-  {
-    // Couldn't crop the region (requested region is outside the largest
-    // possible region).  Throw an exception.
-
-    // store what we tried to request (prior to trying to crop)
-    inputPtr->SetRequestedRegion(inputRequestedRegion);
-
-    // build an exception
-    InvalidRequestedRegionError e(__FILE__, __LINE__);
-    e.SetLocation(ITK_LOCATION);
-    e.SetDescription("Requested region is (at least partially) outside the largest possible region.");
-    e.SetDataObject(inputPtr);
-    throw e;
-  }
-}
-
```

Comment:
The following is the preferred pattern to when a filter must produce all of it's outputs and requires all of it's inputs:
```
EnlargeOutputRequestedRegion(
    DataObject * output)
  {
    Superclass::EnlargeOutputRequestedRegion(output);

    if (output != nullptr)
    {
      output->SetRequestedRegionToLargestPossibleRegion();
    }
  }

```